core/src/com/unciv/ui/tilegroups/TileGroup.kt

```diff
@@ -557,14 +556,17 @@ open class TileGroup(var tileInfo: TileInfo, var tileSetStrings:TileSetStrings,
                 ImageGetter.imageExists(specificUnitIconLocation + "-" + militaryUnit.civInfo.nation.style) ->
                     specificUnitIconLocation + "-" + militaryUnit.civInfo.nation.style
                 ImageGetter.imageExists(specificUnitIconLocation) -> specificUnitIconLocation
-                militaryUnit.baseUnit.replaces!=null &&
+                militaryUnit.baseUnit.replaces != null &&
                         ImageGetter.imageExists(tileSetStrings.unitsLocation + militaryUnit.baseUnit.replaces) ->
                     tileSetStrings.unitsLocation + militaryUnit.baseUnit.replaces
-
-                unitType == UnitType.Mounted -> tileSetStrings.unitsLocation + "Horseman"
-                unitType == UnitType.Ranged -> tileSetStrings.unitsLocation + "Archer"
-                unitType == UnitType.Armor -> tileSetStrings.unitsLocation + "Tank"
-                unitType == UnitType.Siege -> tileSetStrings.unitsLocation + "Catapult"
+    
+                /** FIXME: these should probably not be hardcoded, but I don't know any other easy solution
```

Comment:
I think we can search the units for the first unit of this type that has an image, and take its sprite

Reply:
These are _fallbacks_ in case a specific unit is missing a pixel unit image - I'd at least persist the choice outside of `TileGroup.updatePixelMilitaryUnit` - doesn't that run often? `imageExists` can be surprisingly expensive on Android. And - I'd put it in json, cheap, and if that's missing in e.g. a mod - simply no pixel unit.